Library/Homebrew/service.rb

```diff
@@ -47,6 +47,30 @@ def working_dir(path = nil)
       end
     end
 
+    sig { params(path: T.nilable(T.any(String, Pathname))).returns(T.nilable(String)) }
+    def root_dir(path = nil)
+      case T.unsafe(path)
+      when nil
+        @root_dir
+      when String, Pathname
+        @root_dir = path.to_s
+      else
+        raise TypeError, "Service#root_dir expects a String"
+      end
+    end
+
+    sig { params(path: T.nilable(T.any(String, Pathname))).returns(T.nilable(String)) }
+    def input_path(path = nil)
+      case T.unsafe(path)
+      when nil
+        @input_path
+      when String, Pathname
+        @input_path = path.to_s
+      else
+        raise TypeError, "Service#input_path expects a String"
```

Comment:
Same here (see above comment)

```suggestion
        raise TypeError, "Service#input_path expects a String or Pathname"
```